internal/extramanifest/extramanifest.go

```diff
@@ -79,78 +109,31 @@ func (h *EMHandler) ExportExtraManifestToDir(ctx context.Context, extraManifestC
 	return nil
 }
 
-// ApplyExtraManifestsFromDir applies the extra manifests from the given directory
-func (h *EMHandler) ApplyExtraManifestsFromDir(ctx context.Context, fromDir string) error {
-	manifestDir := filepath.Join(fromDir, extraManifestPath)
-	manifestYamls, err := os.ReadDir(manifestDir)
+// ApplyExtraManifests applies the extra manifests from the preserved extra manifests directory
+func (h *EMHandler) ApplyExtraManifests(ctx context.Context, executor ops.Execute) error {
+	_, err := os.Stat(common.PathOutsideChroot(extraManifestPath))
 	if err != nil {
+		if os.IsNotExist(err) {
+			h.Log.Info("No extra manifests directory found, skipping...", "path", extraManifestPath)
+			return nil
+		}
 		return err
 	}
 
-	for _, manifestYaml := range manifestYamls {
-		if manifestYaml.IsDir() {
-			h.Log.Info("Unexpected directory found, skipping...", "directory",
-				filepath.Join(manifestDir, manifestYaml.Name()))
-			continue
-		}
-
-		manifestBytes, err := os.ReadFile(filepath.Join(manifestDir, manifestYaml.Name()))
-		if err != nil {
-			return err
-		}
-
-		manifest := &unstructured.Unstructured{}
-		if err := yaml.Unmarshal(manifestBytes, manifest); err != nil {
-			return err
-		}
-
-		manifest.SetUID("")
-		manifest.SetResourceVersion("")
-		if err := h.Client.Create(ctx, manifest); err != nil {
-			if !k8serrors.IsAlreadyExists(err) {
-				return err
-			}
-
-			// Update if it already exists
-			existingManifest := &unstructured.Unstructured{}
-			existingManifest.SetGroupVersionKind(manifest.GroupVersionKind())
-			if err := h.Client.Get(ctx, types.NamespacedName{
-				Name:      manifest.GetName(),
-				Namespace: manifest.GetNamespace(),
-			}, existingManifest); err != nil {
-				return err
-			}
-
-			manifest.SetResourceVersion(existingManifest.GetResourceVersion())
-			manifest.SetUID(existingManifest.GetUID())
-			if err := h.Client.Update(ctx, manifest); err != nil {
-				return err
-			}
-			h.Log.Info("Updated manifest", "name", manifest.GetName(), "namespace", manifest.GetNamespace())
-		} else {
-			h.Log.Info("Created manifest", "name", manifest.GetName(), "namespace", manifest.GetNamespace())
-		}
+	h.Log.Info("Applying DU extra manifests from path", "path", extraManifestPath)
+	args := []string{"--kubeconfig", common.KubeconfigFile, "apply", "-f"}
+	_, err = executor.Execute("oc", append(args, extraManifestPath)...)
```

Comment:
can we use the client defined in `ImageBasedReconciler`?

Reply:
Yes, we are using the client in ImageBasedReconciler. In the upgrade handler, `r.Executor` is passed into this function.

Reply:
oh sorry I meant the runtime client. I see this handler has also a client. Can we use that instead of `oc`?

Reply:
I see what you mean! To use runtime client, you will need proper RBAC in order to create the resource. The problem is that LCA doesn't have the knowledge of the KIND of these resources. We cannot assume we will ony have sriovnetwork, sriovnetworknodepolicies etc in the extra manfiests configmap.

Reply:
yes we cannot assume the KIND and LCA should not care, it needs to be agnostic.

Reply:
@Missxiaoguo this looks like job for client-go's out-of-cluster feature see https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go 

Should be a lot faster than using pure exec

Reply:
@pixelsoccupied wouldn't that still have the RBAC requirements?

Reply:
i dont think so! Generally when a kubeconfig is not specified it falls back to InClusterConfig i.e SA...which in-turn needs RBAC. But if we pass KUBECONFIG (like for kubectl or out-of-cluster)...that requirement should be skipped. But please double check!

Reply:
@pixelsoccupied Sure! Let me give a shot!

Reply:
There is https://github.com/openshift/library-go/blob/master/pkg/assets/create/creater.go for such things as far as i remember, at least ocp clusterbootstrap uses this way with admin kubeconfig

Reply:
Got this working with client-go. Will push an update